### ISSUE TITLE
"Customize" installer button does nothing when umbracoDbDSN is specified

### ISSUE THREAD

**Issue by OzoneNZ** [umbraco/Umbraco-CMS#11187]
### Which *exact* Umbraco version are you using? For example: 8.13.1 - don't just write v8

9.0.0-rc004

### Bug summary

When an `umbracoDbDSN` is specified, the installer's `Customize` button has no effect but is still displayed.

The installer simply proceeds as if the `Install` button had been clicked instead.

### Steps to reproduce

**Scenario 1**

```json
{
  "ConnectionStrings": {
    "umbracoDbDSN": ""
  }
}
```

![No connection string specified](https://user-images.githubusercontent.com/4223195/134599185-346c2cb3-aeb5-45f4-9d79-2476e7d59639.gif)

**Scenario 2**

```json
{
  "ConnectionStrings": {
    "umbracoDbDSN": "Data Source=|DataDirectory|\\Umbraco.sdf;Flush Interval=1;"
  }
}
```

![Connection string specified](https://user-images.githubusercontent.com/4223195/134599230-144860b4-9d7b-49c6-8332-7222b75ae3f9.gif)

### Expected result / actual result

Either the `Customize` button should not be displayed when an `umbracoDbDSN` has already been specified, or the `Customize` button should allow the connection string/database to be changed anyway.

**Comment by JoseMarcenaro** [comment]
Hi @OzoneNZ I don't understand the use case for Scenario 2 (the one failing)

Do you want the install process to setup a new Umbraco DB, but pre-assigning the `umbracoDbDSN` in appsettings.json instead of filling the install form?

Using the release version of Umbraco 9.0.0  (August 28th) I tested that scenario with a **valid connection string pointing to an existing (empty) DB** and the installation goes through with no issues - no [ Customize ] button either.  

But maybe that is not what you want to do?

**Comment by OzoneNZ** [comment]
Hey @JoseMarcenaro, the use-case for me is that we ship this connection string in our repository:

```json
{
  "ConnectionStrings": {
    "umbracoDbDSN": "Data Source=|DataDirectory|\\Umbraco.sdf;Flush Interval=1;"
  }
}
```

But we do not ship an `Umbraco.sdf` file, so we want Umbraco to auto-create the SDF file when using the installer for the first time (which it does do correctly).

The problem I have with scenario 2 is that the `Customize` should either **not appear**, or actually allow you to select a different database provider as normal. Right now it's behaving the same as the `Install` button so it serves no purpose.

Its current behavior is very confusing because it implies you can change the database provider, but it just continues using the already-supplied connection string and performs an install.

**Comment by OzoneNZ** [comment]
I should also note, the same behaviour occurs when using a normal `Microsoft SQL Server` connection string; the `Customize` button no longer allows overriding the database provider as soon as an `umbracoDbDSN` is provided

**Comment by JoseMarcenaro** [comment]
Hi @OzoneNZ  I just performed this sequence with `9.0.0`

1. Create new website from scratch (using dotnet new ...etc)
2. Assign a valid LocalDB connection string (*) in the "umbracoDbDSN" property of appsettings.json
3. Run the website

And it worked - no questions asked, no "Customize" - it created the umbraco.mdf database file and installed Umbraco normally.

Maybe the issue you are having is that the connection string format is no longer valid for .Net 5 and/or the recent versions of LocalDB.

Please check this MS page: https://docs.microsoft.com/en-us/sql/database-engine/configure-windows/sql-server-express-localdb?view=sql-server-ver15#start-localdb-and-connect-to-localdb

The exact connection string I used was:
```
"umbracoDbDSN": "Server=(LocalDB)\MSSQLLocalDB;AttachDbFilename=|DataDirectory|\\Umbraco.mdf;Integrated Security=True"
```
Let me know if this solved your problem.

**Comment by JoseMarcenaro** [comment]
By the way, if you run a clean v9.0.0 installation (no changes in the connection string), 
and keep the default choice "Microsoft SQL Server Express (LocalDB)", when the installation completes this is the connection string you get:
```
"umbracoDbDSN": "Data Source=(localdb)\\MSSQLLocalDB;AttachDbFilename=|DataDirectory|\\Umbraco.mdf;Integrated Security=True"
```
(the terms "Server" and "Data Source" are interchangeable)

**Comment by nul800sebastiaan** [comment]
Yes, this is a valid issue, if you provide a connection string then the installer should not offer you to Customize (and therefore pretend you can change the connection string). We'd be happy for some help here! 👍

**Comment by nul800sebastiaan** [comment]
Fixed for 9.1.0 in https://github.com/umbraco/Umbraco-CMS/pull/11273
